src/pages/singleProduct/SingleProduct.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import './SingleProduct.css'
+import { fetchSingleProduct } from 'helpers/singleProduct/fetchSingleProduct';
+import { useCart } from 'context/addToCart';
+import { useWishlist } from 'context/addToWishlist';
+
+function SingleProduct() {
+    const [, dispatchCart] = useCart();
+    const [, dispatchWishlist] = useWishlist();
+    const [singleproduct, setSingleProduct] = useState(null)
+    const [itemWishlisted, setItemWishlisted] = useState(false)
+    const { productID } = useParams();
+    useEffect(() => {
+        (async () => {
+            try {
+                if (productID) {
+                    const response = await fetchSingleProduct(productID)
+                    setSingleProduct(response.product)
+                }
+            } catch (e) {
+                console.error(e)
+            }
+        })()
+    }, [])
```

Comment:
You can also get the particular product by filtering over the existing products array.